benchmark/src/jvmMain/kotlin/benchmark/Benchmark.kt

```diff
@@ -0,0 +1,164 @@
+@file:Suppress("unused")
+
+package benchmark
+
+import de.westnordost.osm_opening_hours.model.OpeningHours
+import de.westnordost.osm_opening_hours.parser.toOpeningHoursOrNull
+import dev.sargunv.kotlindsv.Tsv
+import kotlinx.benchmark.Benchmark
+import kotlinx.benchmark.BenchmarkMode
+import kotlinx.benchmark.BenchmarkTimeUnit
+import kotlinx.benchmark.Mode
+import kotlinx.benchmark.OutputTimeUnit
+import kotlinx.benchmark.Scope
+import kotlinx.benchmark.Setup
+import kotlinx.benchmark.State
+import kotlinx.benchmark.TearDown
+import kotlinx.io.buffered
+import kotlinx.io.files.Path
+import kotlinx.io.files.SystemFileSystem
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class InputRecord(
+    @SerialName("?times") val times: String,
+    @SerialName("?count") val count: Int,
+)
+
+class ResultRecord {
+    var parsed: OpeningHours? = null
+    var created: String? = null
+    var valid: Boolean = false
+    var pooleParsed: List<ch.poole.openinghoursparser.Rule>? = null
+    var pooleCreated: String? = null
+    var pooleValid: Boolean = false
+}
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(BenchmarkTimeUnit.MILLISECONDS)
+open class Benchmark {
+    lateinit var state: List<Pair<InputRecord, ResultRecord>>
+
+    @Setup
+    fun prepare() {
+        state =
+            SystemFileSystem.source(Path("../sample-data/opening_hours_counts.tsv")).buffered().use {
+                source ->
+                Tsv.decodeFromSource<InputRecord>(source).map { it to ResultRecord() }.toList()
+            }
+    }
+
+    @Benchmark
+    fun parse() = state.forEach { (input, result) ->
+        input.times.toOpeningHoursOrNull(lenient = true)?.let { oh -> result.parsed = oh }
+    }
+
+    @Benchmark
+    fun parseStrict() = state.forEach { (input, result) ->
+        input.times.toOpeningHoursOrNull(lenient = true)?.let { _ -> result.valid = true }
```

Comment:
```suggestion
        result.valid = input.times.toOpeningHoursOrNull(lenient = false) != null
```

Bug: strict parsing, so `lenient = false`